### PR TITLE
Length constants for username and password checking.

### PR DESCRIPTION
**Pull request by andorp**
Fixes: #632

### CODE REVIEW

File: src/Frontend/Validation.hs [hunk 1]
```diff
@@ -48,9 +48,10 @@ import Text.Email.Validate as Email
 import Text.Parsec as TP hiding (Reply(..))
 import Text.Parsec.Error
 
+import Frontend.Constant
 import Frontend.Prelude as Frontend hiding ((<|>))
 
-import qualified Data.Text as ST 
+import qualified Data.Text as ST
```
Comment (fisx) [inline src/Frontend/Validation.hs]: thanks (:

File: src/Frontend/Validation.hs [hunk 2]
```diff
@@ -207,10 +208,15 @@ usernameV' = dimap (view _UserLogin) UserLogin usernameV
...
+usernameV = fieldParser
+    (cs <$> manyNM minUsernameLength maxUsernameLength letter
+            <??> concat [ show minUsernameLength, "-"
+                        , show maxUsernameLength, " Buchstaben"])
 
 passwordV :: StringFieldValidator
-passwordV = fieldParser (cs <$> manyNM 4 12 anyChar <??> "4-12 Zeichen")
+passwordV = fieldParser
+    (cs <$> manyNM minPasswordLength maxPasswordLength anyChar
+            <??> (show minPasswordLength <> "+ Zeichen"))
```
Comment (fisx) [inline src/Frontend/Validation.hs]: why is the max length not shown in the error?

**Comment by fisx** [comment]
LGTM, just one comment.

File: src/Frontend/Validation.hs (hunk 2 above)
Comment (andorp) [inline src/Frontend/Validation.hs]: You are right, fixed in 1690034

**Comment by fisx** [comment]
this will probably break the test...  (-:

File: tests/Frontend/Page/LoginSpec.hs [hunk 3]
```diff
@@ -12,7 +13,7 @@ spec = describe "logging in" $ do
     context "if user does not exist" $ do
       it "will not log you in and will redirect" $ \wreq -> do
         -- FIXME: Figure out what went wrong.
-        pendingWith "Quote is in the username... Login : unerwartet &quot; &quot;erwartet 4-12 Buchstaben"
+        pendingWith $ "Quote is in the username... Login : unerwartet &quot; &quot;erwartet " <> show minPasswordLength <> "+ Zeichen"
```
Comment (fisx) [inline tests/Frontend/Page/LoginSpec.hs]: ah, i was confused by this string, but it's a `pendingWith` note.

**Comment by fisx** [comment]
thanks!

File: tests/Frontend/Page/LoginSpec.hs (hunk 3 above)
Comment (andorp) [inline tests/Frontend/Page/LoginSpec.hs]: Me too at the first time :)

**Pull request merged**